types/src/transaction/transaction.rs

```diff
@@ -754,4 +862,22 @@ mod tests {
         };
         rlp_encode_and_decode_test!(transaction);
     }
+
+    #[test]
+    fn fail_if_filter_is_illegal() {
+        let transaction = Transaction::AssetTransfer {
+            network_id: NetworkId::default(),
+            burns: Vec::new(),
+            inputs: Vec::new(),
+            outputs: Vec::new(),
+            nonce: 0,
+        };
+        let outpoint = AssetOutPoint {
+            transaction_hash: H256::default(),
+            index: 0,
+            asset_type: H256::default(),
+            amount: 0,
+        };
+        assert_eq!(transaction.hash_partially(5 as u128, &outpoint, false), Err(HashingError::InvalidFilter));
```

Comment:
`0b101` instead of 5.
Using binary format is better since it's a bitmask.

Reply:
I resolved it